nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -163,6 +163,7 @@ NavfnPlanner::computePathToPose(const std::shared_ptr<GoalHandle> goal_handle)
 
   try {
     // Get the current pose from the robot
+    getCostmap(costmap_);
```

Comment:
Instead of directly setting the new costmap to our local copy `costmap_`, we should check that the size of the new costmap is the same. If it is, then it's safe to replace our local copy, but if not, then return an abort.

Reply:
On lifecycle-less version we would create a new core navfn planner on the fly with these new costmap sizes. During creation, the core navfn planner allocates resources based on the size of the costmap. Now on the lifecycle version we need to decide what to do, I think ideally we want to keep the creation process on the `on_configure` state, and expect that the costmap size will not change, is this a correct assumption?

If we go the abort route, it would be useful to report the failure mode so that actions can be taken. For example, `bt_navigator` could try a recovery action, in this case it means taking the planner back to `on_configure`. This could alternatively be done by reporting the fail mode to `lifecycle_manager`.

Just some ideas for discussion, for now, my proposal is to add the check and report abort.

Reply:
I don't think aborting is the right way to go. Changing costmap sizes seem like a normal operating mode. We just need to manage the memory/reconfiguration in a real-time friendly way. But for now, just using the existing mechanism is the way to go.

Reply:
To do that, you also need this section of my PR
```
 if (isPlannerOutOfDate()) {
      current_costmap_size_[0] = costmap_.metadata.size_x;
      current_costmap_size_[1] = costmap_.metadata.size_y;
      planner_ = std::make_unique<NavFn>(costmap_.metadata.size_x, costmap_.metadata.size_y);
    }
   }
```

Reply:
I also don't think we should abort, but I don't think we need to re-create the planner but just reallocate the the memory using existing functionality: 
```suggestion
    if (isPlannerOutOfDate()) {
      current_costmap_size_[0] = costmap_.metadata.size_x;
      current_costmap_size_[1] = costmap_.metadata.size_y;
      planner_->setNavArr(costmap_.metadata.size_x, costmap_.metadata.size_y);
    }
```

Reply:
We don't get a huge benefit from just calling `setNavArr` over recreating the whole planner. There are many memory deletes and reallocations. 

How sure are you that none of that none of the variables set in the constructor need to be reinitialized?

Reply:
The [constructor]( https://github.com/ros-planning/navigation2/blob/955fc29dc63467de5a3b6dc5aaafd8d1207871fd/nav2_navfn_planner/src/navfn.cpp#L105) calls `setNavArr` and all other variables are set to constant values, so I saw no reason to recreate the whole planner.